src/lexitools/commands/correspondences.py

```diff
@@ -2,85 +2,150 @@
 Check the prosodic structure of a given dataset.
 """
 
-from clldutils.clilib import Table, add_format
-from cldfbench.cli_util import add_catalog_spec, get_dataset
+from clldutils.clilib import add_format
+from cldfbench.cli_util import add_catalog_spec
 
-# from linse.transform import syllable_inventories
-from pylexibank.cli_util import add_dataset_spec
 from pyconcepticon.api import Concepticon
 from cldfcatalog import Config
 
 # lingpy for alignments
 import lingpy
-# need this to store as GML format (for inspection)
-from lingpy.convert.graph import networkx2igraph
 
 from itertools import combinations, product
 from collections import Counter, defaultdict
-import matplotlib.pyplot as plt
 from pylexibank import progressbar
-
+import pyglottolog
 import networkx as nx
 import csv
 import csvw
-
-class DataByGenera(object):
-
-    def __init__(self, ds, langgenera_path, filter=None, sound_class=None):
-        if filter is None:
-            filter = lambda *args: True
+import time
+import subprocess
+from importlib.util import find_spec
+from cldfbench import get_dataset
+
+
+class MockLexicore(object):
+    """This is a mock to access the data without waiting on the Lexibank SQL project,
+    but with a reasonable interface so that we can plug in the correct interface soon with little effort."""
+
+    def __init__(self, dataset_list):
+        # see https://github.com/chrzyki/snippets/blob/main/lexibank/install_datasets.py
+        egg_pattern = "git+https://github.com/lexibank/{name}.git#egg=lexibank_{name}"
+        for name in dataset_list:
+            if find_spec("lexibank_" + name) is None:
+                subprocess.run(
+                    ["pip", "install", "-e", egg_pattern.format(name=name)])
+        self.datasets = {}
+        for name in dataset_list:
+            try:
+                self.datasets[name] = get_dataset(name,
+                                                  ep="lexibank.dataset").cldf_reader()
+            except AttributeError:
+                print("Failed to load", name)
+
+    def iter_table(self, table_name):
+        for name, ds in self.datasets.items():
+            for row in ds[table_name]:
+                row["dataset"] = name
+                yield row
+
+    def get_table(self, table_name):
+        return list(self.iter_table(table_name))
```

Comment:
This is what simulates the lexicore interface. Given a list of datasets, it simply installs each dataset (if not already installed)  and then chains the dataset tables together. This should be very similar to the interface @tresoldi wrote for the python code to load the sql database from lexicore.

Reply:
Yes, it is exactly like the interface in `lexicore.py`.